worker-jobtracking/src/main/java/com/hpe/caf/worker/jobtracking/JobTrackingWorkerFactory.java

```diff
@@ -389,4 +393,158 @@ private void forwardAvailableJobs(final List<JobTrackingWorkerDependency> jobDep
             LOG.error("Error retrieving Dependent Job Info from the Job Service Database {}", e);
         }
     }
+
+    @Override
+    public void processTasks(final BulkWorkerRuntime bwr) throws InterruptedException
+    {
+        // Configured to wait 10 seconds between two batches
+        final long maxWaitingTime = Long.getLong(System.getenv("MAX_BATCH_WAITING_TIME"));
+        final long cutoffTime = System.currentTimeMillis() + maxWaitingTime;
+
+        final HashMap<WorkerTaskBulkItem, List<String>> bulkItemList = new HashMap<>();
```

Comment:
You have included `WorkerTask` in the key.  You're just trying to group by job so only the partition id and job id should be included in the key.  The value of the map should be `List<WorkerTask>`.

Reply:
done